x-pack/plugins/lists/common/schemas/common/schemas.ts

```diff
@@ -216,7 +217,11 @@ export type _Tags = t.TypeOf<typeof _tags>;
 export const _tagsOrUndefined = t.union([_tags, t.undefined]);
 export type _TagsOrUndefined = t.TypeOf<typeof _tagsOrUndefined>;
 
-export const exceptionListType = t.keyof({ detection: null, endpoint: null });
+export const exceptionListType = t.keyof({
+  [ENDPOINT_TRUSTED_APPS_LIST_TYPE]: null,
```

Comment:
@madirey , @marshallmain ,
There are some check/test failures due to this change. Before I go down the path of adjusting those to account for this new value, I just want to confirm that this is the correct change to do here.

Reply:
The reason I thought you might want a different list type if because of things like this: https://github.com/elastic/kibana/blob/88c063134464047c5da471e480dd26cabe471a02/x-pack/plugins/lists/server/routes/create_exception_list_item_route.ts#L75-L90 ... if that check is valid, and you don't need any additional checks, you might get away with using type `endpoint`. But it seems like you might want your own type. If you add a different type, and your own route, I think you should add validation to that route so that requests to create your list type are rejected.

@yctercero do you have any input on this?

Reply:
This is one of the reasons why I think this gets weird if we use a different route to manage the list, rather than having it managed by the `lists` plugin, btw... if you add your own route, you still will need validation to ensure that `lists` can't work with your list type.

Reply:
I'm going revert this type back and only introduce the method to create the trusted apps list. We'll revisit later when we have more clarity. Perhaps we just add the new type to the exceptions service methods in support of Trusted Apps? Maybe? 😄

Reply:
Question: since this validation you referenced is in the exceptions list route, I don't think it matters to trusted apps - right?

(Still agree that we should introduce code to prevent attempts to update a Trusted Apps list item in these APIs)

Reply:
The problem is [here](https://github.com/elastic/kibana/pull/74868/files#diff-95fb9bb72aed741f874fb1ef04e28115R60) we shouldn't use `endpoint` as the type - the trusted apps list is not the same as the endpoint exceptions lists. When we create the trusted apps list it needs to have its own type so I think the type has to be added with the method to create the list. The type needs to be included in `exceptionListSoSchema` for the saved object creation type check to pass, so I think it's logical to include the additional type here. We do need additional validation to prevent adding trusted apps through the generic `create_exception_list_item` route - for now the simplest approach would be to just add another check like `if(exceptionList.type == TRUSTED_APPS_LIST_TYPE) return 400`.

We may want to consider again making each type of list use their own route so that the route handler knows what type of list to expect and can immediately return 400 if a request attempts to add to a different type of list. This makes the handlers more independent of each other - create_exception_list_item_route would only be responsible for checking that you're adding the item to a `detection` exception list and doesn't care what other types of lists may exist.

Reply:
I'll create a separate issue to introduce a new type for use by Trusted apps.  I attempted on this one, but looks like it had impacts not only to the schemas, but also types used on the Detections Engine (exceptions?) UI. We'll come back around to it.